phpunit/blocks/render-query-test.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * Tests for the Query block rendering.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ * @since 6.0.0
+ *
+ * @group blocks
+ */
+class Tests_Blocks_RenderQueryBlock extends WP_UnitTestCase {
+
+	private static $post_1;
+	private static $post_2;
+	private static $post_3;
+
+	public function set_up() {
+		parent::set_up();
+
+		self::$post_1 = self::factory()->post->create_and_get(
```

Comment:
@DAreRodz, can you take a look at this? 🙂

Reply:
Since only Post IDs seem to be needed here, use `create()` instead of `create_and_get()`. Event better, use `create_many()` and store them all in one class property. Especially since `$post_1` and `$post_3` are not directly referenced anyway.

Reply:
Thanks for the insights, folks. As @darerodz and I are not very experienced with PHP unit tests yet, we will tackle these improvements in a subsequent PR to avoid blocking this one 🙂

Resolving this to keep things clear.

Reply:
I just created this PR to apply the suggested improvements:

- https://github.com/WordPress/gutenberg/pull/55805

Reply:
Thanks, David!